src/tir/transforms/common_subexpr_elim_tools.cc

```diff
@@ -727,7 +728,10 @@ bool EquivalentTerms(const PrimExpr& a, const PrimExpr& b) {
   // For now, we just check the syntactic equality, but that could later become a semantic test,
   // for instance identifying computations modulo commutativity (like x+y and y+x), or modulo
   // associativity (like (x+y)+z and x+(y+z)), etc.
-  return EqualTerms(a, b);
+  arith::Analyzer analyser;
```

Comment:
Thanks @yuanfz98 , One thing to note here is that running recursively simplification on every subexpression would result in a possibly quadratic complexity wrt to the expression size, so we would want to use it with care, perhaps only triger with limited expression length to avoid long compilation time.

In this particular case, directly running a simplification pass before the common subexpr elim would have a same effect.

Reply:
Yes I agree, I will close this PR. I will follow the ongoing development of pass dependencies and come back to this feature after. Thanks for your reply ! 
cc @FranckQC @zhiics